### DISCUSSION TITLE
Controlling the `air.` prefix on Android applications

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#1111] [Q&A]
I'm currently implementing some utilities in [`apm`](https://github.com/airsdk/apm) around application descriptor generation and I need to determine whether to prefix the application id with `air` to generate the android package name. 

What are all the current methods to disable the default behaviour? 

- Environment variable `AIR_NOANDROIDFLAIR=true`
- `adt.cfg` set `AddAirToAppID=false` 
- Was there a command line option added ? 

Is there an order to these, ie which ones override others ? 

I was hoping to be able to determine this automatically but I feel it's not going to be possible with all these different approaches that have been added. 

How are people using these options in their builds?

**Comment by ajwfrost** [comment]
In terms of the `air` prefix:
1) yes those are the only two mechanisms, there's no command-line option in ADT for this (I think we have a task low down in the backlog for it still....)
2) the "add prefix" setting is on by default, but can be turned off by either the config file or the environment variable. These each only have an affect if they would turn off the "add prefix" setting, i.e. if they are missing or set to "add prefix", they are just ignored. So if either mechanism is saying "don't add the prefix", then the prefix isn't added, even if the other setting is saying it should be added. 
3) we appear to use "air.packagename.provider" as the authorities attribute for any provider elements listed in the manifest..?! so I would be interested to know if that means it fails to work if you don't allow the prefix..

Personally I don't think we should be adding the 'air.' prefix any more. I think it was done originally just for marketing/analytics purposes, and it is interesting to see what apps are developed in AIR judging by the app ID, but it's only done on Android and there's always been this opt-out thing .. I don't think there's any particular benefit to it now.

But of course -> we shouldn't change it for existing apps. (and now I see your point re these being app-specific settings...!)

thanks

**Reply by marchbold** [reply]
Okay, so if I can read the environment variable and the adt.cfg file in the air sdk I should be able to guess at what adt will do. (glad there is no command line option as that would have made it impossible).

Might have to have a think about how I can access that `adt.cfg` file.... Don't think that's going to be easy....

And yes, this one is such a long standing expectation, you'll have to make a lot of noise and make it simple to specify it per project if you ever decide to change the default operation here (or even if you don't it would be good). But I totally agree the prefix is unnecessary.

**Reply by ajwfrost** [reply]
If you're launching the APM app via ADL then you could pass in another parameter to it which is the executable path of ADL -> then resolve ../lib/adt.cfg to get the file? It's a JavaScript 'Properties' format, which seems to be similar to a Windows .ini file..

**Reply by marchbold** [reply]
Ah yep of course, I should be able to do something like that, shouldn't be working so late :)
